presto-main/src/main/java/io/prestosql/sql/planner/LocalDynamicFilterConsumer.java

```diff
@@ -195,9 +195,7 @@ public static LocalDynamicFilterConsumer create(JoinNode planNode, List<Type> bu
     private static boolean isFilterAboveTableScan(PlanNode node)
     {
         if (node instanceof FilterNode) {
-            if (((FilterNode) node).getSource() instanceof TableScanNode) {
-                return true;
-            }
+            return ((FilterNode) node).getSource() instanceof TableScanNode;
         }
         return false;
```

Comment:
return node instanceof FilterNode && ((FilterNode) node).getSource() instanceof TableScanNode;

Reply:
I found it has been done in https://github.com/prestosql/presto/pull/4300